docs/testing-with-phpunit.md

```diff
@@ -0,0 +1,225 @@
+# Testing With PHPUnit
+
+[PHPUnit]() is a powerful framework for testing PHP code and applications. The tests that PHPUnit excels at running can be broken down into two fundamental types:
+
+- Unit tests
+- Integration tests
+
+Unit tests can be run against a unit of code in isolation, such as a function or object and require no persistent state.
+
+Integration tests allow you to test a running application. In the case of Altis this type of testing is often more useful however it is possible to run both types of test.
+
+Currently only PHPUnit <= 7.1 is supported.
+
+
+## Setting Up PHPUnit
+
+To set up PHPUnit from scratch run the following command:
+
+```sh
+composer dev-tools scaffold phpunit
+```
+
+This will create the following files:
+
+- `phpunit.xml.dist` - Instructions for PHPUnit.
+- `.tests/bootstrap.php` - Handles loading and installing Altis before running tests.
```

Comment:
We encourage this in the handbook: https://engineering.hmn.md/standards/structure/ -Using a `.` ensures that it's not accidentally exposed on web servers if deployed.

This is only for the meta stuff around running tests though, not the tests themselves.

Reply:
I've been basing this on observations from our existing projects, so often there's a tests directory containing both php & js tests in their own subfolders. As I've removed this scaffolding now perhaps it could be made configurable.

Reply:
Can I ask why you prefer to avoid placing the two types of tests in the same directory? My feeling is it that JS tests could go inside the `src` directory, but it makes sense for there to be a `tests` directory in the project root alongside PHP files. Re. the name I can see `wp-tests` is being very explicit but in the context of what we do I'm not sure what the value is unless you're saying you'd have a `wp-tests` and a `js-tests` file at the same level.

I'll keep it simple for now, either `tests` or `.tests` is supported by default. Lets see what the beta release feedback is like.